packages/ia-userlist-settings/src/user-lists-service/user-lists-service.ts

```diff
@@ -0,0 +1,236 @@
+import type { Result } from '@internetarchive/result-type';
+import {
+  SearchType,
+  SearchResult,
+  SearchServiceInterface,
+} from '@internetarchive/search-service';
+import type { UserListsServiceInterface } from './user-lists-service-interface';
+import type {
+  UserList,
+  UserListMember,
+  UserListMemberOptions,
+  UserListOptions,
+} from './models';
+import { UserListsErrorReason, UserListsError } from './user-lists-error';
+import type { UserServiceInterface } from '@internetarchive/user-service';
+
+/** Required shape of the fetch handler provided to the service */
+interface FetchHandlerInterface {
+  fetchApiResponse<T>(
+    url: string,
+    options?: {
+      includeCredentials?: boolean;
+      method?: string;
+      body?: BodyInit;
+    },
+  ): Promise<T>;
+}
+
+/** Possible well-formed results from the backend service */
+type BackendServiceResponse<T> =
+  | { success: true; value: T }
+  | { success: false; error: string };
+
+/** Options accepted by the user lists service constructor */
+type UserListsServiceConstructorOptions = {
+  fetchHandler: FetchHandlerInterface;
+  searchService: SearchServiceInterface;
+  userService: UserServiceInterface;
+  baseUrl?: string;
+};
+
+/** Service for fetching forum posts */
+export class UserListsService implements UserListsServiceInterface {
+  private static readonly DEFAULT_BASE_URL = 'https://archive.org';
+
+  private fetchHandler: FetchHandlerInterface;
+  private searchService: SearchServiceInterface;
+  private userService: UserServiceInterface;
+  private baseUrl: string;
+
+  constructor({
+    fetchHandler,
+    searchService,
+    userService,
+    baseUrl = UserListsService.DEFAULT_BASE_URL,
+  }: UserListsServiceConstructorOptions) {
+    this.fetchHandler = fetchHandler;
+    this.searchService = searchService;
+    this.userService = userService;
+    this.baseUrl = baseUrl;
+  }
+
+  private async fetchEndpoint<T>(
+    url: string,
+    method?: string,
+    body?: BodyInit,
+  ): Promise<Result<T, UserListsError>> {
+    try {
+      const fetchResult: BackendServiceResponse<T> = await this.fetchHandler.fetchApiResponse(
```

Comment:
How the `this.fetchHandler` being used in this repo? are you creating new handler we already in offshoot repo?

Reply:
Good question -- plan is to extract Offshoot's fetch handler into iaux as well, so that it can be used here. However, the interface here only requires the one `fetchApiResponse` method and is not tied to a specific implementation, so in theory one could just pass in a simple handler that just delegates straight to the browser's `fetch`.